### PR TITLE
swift-protobuf 1.0.0 (new formula)

### PR DESCRIPTION
**Pull request by anderscarling**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Apple's swift plugin for protobuf. Notably, this conflicts with `protobuf-swift` which also has a confusingly similar name. Maybe we should prefix name of this formulae with `apple-` to clarify?

### CODE REVIEW

**Comment by ilovezfs** [comment]
Is there some connection between the two code bases?

**Comment by anderscarling** [comment]
I don't think they are connected at all, and seem to be developed independently as far as I can tell.

As far as I can tell they represent two different solutions to the same problem (that is, Protobuf in Swift).

**Comment by anderscarling** [comment]
@ilovezfs Is there any other information you want me to provide? :)

File: Formula/swift-protobuf.rb
Comment (fxcoudert) [inline Formula/swift-protobuf.rb]: No `revision` number, please

File: Formula/swift-protobuf.rb
Comment (fxcoudert) [inline Formula/swift-protobuf.rb]: No man page, doc, or anything else?

File: Formula/swift-protobuf.rb
Comment (fxcoudert) [inline Formula/swift-protobuf.rb]: `(testpath/"test.proto").write <<-EOS.undent`

File: Formula/swift-protobuf.rb
Comment (fxcoudert) [inline Formula/swift-protobuf.rb]: `system "#{bin}/protoc"`

Comment (fxcoudert) [inline Formula/swift-protobuf.rb]: Can we test that the output exists?

**Review by fxcoudert — DISMISSED** [review]

File: Formula/swift-protobuf.rb
Comment (anderscarling) [reply Formula/swift-protobuf.rb]: Sure, I'll add that. Current test is exactly the same as for `protobuf-swift` formulae.

File: Formula/swift-protobuf.rb
Comment (anderscarling) [reply Formula/swift-protobuf.rb]: No man file, but I guess we could `doc.install`  [Documentation/PLUGIN.md](https://github.com/apple/swift-protobuf/blob/master/Documentation/PLUGIN.md) which explains what's how swift-protobuf is used.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/swift-protobuf.rb
Comment (anderscarling) [reply Formula/swift-protobuf.rb]: It actually refers to the `protoc` binary that's supplied from the `protobuf` formula, which in turns uses the `protoc-gen-swift` binary from this formula to generate it's output.

File: Formula/swift-protobuf.rb
Comment (anderscarling) [reply Formula/swift-protobuf.rb]: fixed

**Comment by anderscarling** [comment]
@ilovezfs I made changes based on @fxcoudert's review, let me know if all is good and I'll squash them into the original commit.

**Comment by ilovezfs** [comment]
Still not happy with the naming situation.

**Comment by anderscarling** [comment]
Ok. Short of prefixing this formula with “apple-“, I’m not sure that can be done without making it very hard to find though.

Any other ideas?

-
Anders Carling

**Comment by anderscarling** [comment]
*what can be done

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/protobuf-swift.rb
Comment (cc941201) [inline Formula/protobuf-swift.rb]: `conflicts_with "swift-protobuf"`?

**Review by guoye-zhang — CHANGES_REQUESTED** [review]

File: Formula/protobuf-swift.rb
Comment (anderscarling) [reply Formula/protobuf-swift.rb]: That's very much correct, fixing.

File: Formula/swift-protobuf.rb
Comment (DomT4) [inline Formula/swift-protobuf.rb]: I think this would _(just about)_ be under 80 characters wide if on a single line? If that's the case it's probably fine not to split onto two lines like this.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Review by ilovezfs — CHANGES_REQUESTED** [review]
For Xcode 9, this needs to pass `--disable-sandbox` to `swift build` to avoid the build failure.

See `sourcery` for an example.

**Comment by ilovezfs** [comment]
```
==> swift build -c release -Xswiftc -static-stdlib
error: manifest parse error(s):
sandbox-exec: sandbox_apply_container: Operation not permitted
```

File: Formula/swift-protobuf.rb
Comment (DomT4) [inline Formula/swift-protobuf.rb]:
```ruby
assert_predicate testpath/"test.pb.swift", :exist?
```
?

File: Formula/swift-protobuf.rb
Comment (anderscarling) [reply Formula/swift-protobuf.rb]: It's exactly 80 chars wide, but I think this reads cleaner.

Let me know if projects preferred syntax is single line and I can join them.

File: Formula/swift-protobuf.rb
Comment (anderscarling) [reply Formula/swift-protobuf.rb]: Sure!

**Comment by anderscarling** [comment]
Pushed changes to adress xcode 9, bump to latest version (0.9.905) and fixing test syntax.

Squashed version is available at https://github.com/anderscarling/homebrew-core/tree/swift-protobuf-squashed

File: Formula/swift-protobuf.rb
Comment (DomT4) [reply Formula/swift-protobuf.rb]: Will defer to ILZ. My personal preference is single line where it fits inside the 80, but ILZ runs the show around here so it's his choice.

Comment (ilovezfs) [reply Formula/swift-protobuf.rb]: Yes 1 line if <= 80.

Comment (anderscarling) [reply Formula/swift-protobuf.rb]: Fixed!

Also splitted "swift build" line into two, as it ran a bit longer than 80 chars. Followed the wrapping style from the  `sourcery` formula for that one.

**Comment by ilovezfs** [comment]
@anderscarling do we know what the deal is with this being marked "pre-release" by upstream? Usually that would mean we need to hold off distributing this in Homebrew until it's marked "release" not "pre-release."

**Comment by anderscarling** [comment]
@ilovezfs 
My interpretation of this is that it is in practice more of a pre-1.0 than pre-release.

The library side of the project is published on [cocoapods](https://cocoapods.org/pods/SwiftProtobuf) with no note about it being pre-release, and they do offer full instructions for installing it using a variety of other methods.

There actually was a warning in the readme about the project being pre-release software, but that was removed as they released version [0.9.901](https://github.com/apple/swift-protobuf/commit/a61dbf096a7251ad1cfc1af7da844a3ce20b73a2#diff-04c6e90faac2675aa89e2176d2eec7d8).

I do think it's a useful addition to homebrew at this point. Especially since the client library seems to get a healthy amount of use,  according to cocoapods statistics about 11000 downloads per week - comparable to protobuf-swift's 13000 downloads/week.

The swift-protobuf README currently says they are converging on a 1.0 in the near future though, so I can see why if you would want to wait for that before merging. Maybe @tbkka who seems to be doing the releasing and tagging over at apple/swift-protobuf are able to give some more insight?

**Comment by tbkka** [comment]
We just tagged SwiftProtobuf 1.0.

**Comment by anderscarling** [comment]
I've updated this branch ([and the squashed version](https://github.com/anderscarling/homebrew-core/tree/swift-protobuf-squashed)) to the 1.0.0 release.

**Comment by NachoSoto** [comment]
Any update on this? I'd like to change my setup scripts to install this via Homebrew instead 🙏

**Comment by anderscarling** [comment]
@NachoSoto I think we are basically waiting for final verdict from @ilovezfs

If there is anything more I can do to make this happen, please let me know and I’ll get right on it! :)

File: Formula/swift-protobuf.rb
Comment (DomT4) [inline Formula/swift-protobuf.rb]: This will likely need changing to the squiggly format now. `<~EOS` IIRC.

Comment (DomT4) [reply Formula/swift-protobuf.rb]: It's a very recent change but I presume it's going to get enforced on open PRs.

**Review by ilovezfs — APPROVED** [review]

File: Formula/swift-protobuf.rb
Comment (ilovezfs) [reply Formula/swift-protobuf.rb]: (no longer fits on one line because I added the word "binaries" to be consistent with other conflicts_with statements)

**Comment by ilovezfs** [comment]
Thanks for the new formula @anderscarling and thanks @tbkka for the heads up about the official release. Shipped!

File: Formula/swift-protobuf.rb
Comment (DomT4) [reply Formula/swift-protobuf.rb]: Heh, was just about to check my curiosity on this 😄. Thanks for adding the notation here.

**Pull request closed without merging**
